tests/Model/BSONDocumentTest.php

```diff
@@ -23,6 +23,15 @@ public function testConstructorDefaultsToPropertyAccess(): void
         $this->assertSame('bar', $document->foo);
     }
 
+    public function testConstructorWithStandardObject(): void
+    {
+        $object = (object) ['foo' => 'bar'];
+        $document = new BSONDocument($object);
+        $this->assertEquals(ArrayObject::ARRAY_AS_PROPS, $document->getFlags());
+        $this->assertSame('bar', $document->foo);
```

Comment:
I think these two lines can be omitted, as they seem redundant alongside `testConstructorDefaultsToPropertyAccess()` above. All you really need to test is that a stdClass class can yield the expected `bsonSerialize()` output.